File: DOTNET-ANGULAR/frontend/src/app/_interfaces/ProfileDTO.ts

import { CardDTO } from "./CardDTO";
import { SocialDTO } from "./SocialDTO";

export interface ProfileDTO {
    id: string;
    IsActive: boolean;
    TemplateIndex: number;
    FirstName: string;
    LastName: string;
    PhoneNumber: string;
    Address: string;
    CreatedAt?: Date | null;
    UpdatedAt?: Date | null;
    AccountId: string;
    Cards?: CardDTO[] | null;
    Socials?: SocialDTO[] | null;
    ImagePath?: string | null;
    Latitude: number;
    Longitude: number;
  }

File: DOTNET-ANGULAR/backend/DTO/ProfileDTO.cs
using Microsoft.AspNetCore.Http;
using nfcbackend.Models;

namespace nfcbackend.DTO
{
    public class ProfileDTO:Base
    {
        public bool IsActive { get; set; }
        public int TemplateIndex { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string PhoneNumber { get; set; }
        public string Adress { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public Guid? AccountId { get; set; }
        public List<CardDTO>? Cards { get; set; }
        public List<SocialDTO>? Socials { get; set; }
        
        public string ImagePath { get; set; }
        public float Lattitude { get; set; }
        public float Longitude { get; set; }



    }
}
